src/Framework/Analyzer/test/WebApplicationBuilder/DisallowConfigureHostBuilderConfigureWebHostTest.cs

```diff
@@ -0,0 +1,115 @@
+{
+    private TestDiagnosticAnalyzerRunner Runner { get; } = new(new WebApplicationBuilderAnalyzer());
+
+    [Fact]
+    public async Task WebApplicationBuilder_HostWithoutConfigureWebHost_Works()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+var builder = WebApplication.CreateBuilder();
+builder.Host.ConfigureServices(services => { });
+";
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source);
+
+        // Assert
+        Assert.Empty(diagnostics);
+    }
+
+    [Fact]
+    public async Task WebApplicationBuilder_HostWithConfigureWebHost_ProducesDiagnostics()
+    {
+        // Arrange
+        var source = TestSource.Read(@"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+var builder = WebApplication.CreateBuilder();
+builder.Host./*MM*/ConfigureWebHost(webHostBuilder => { });
+");
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source.Source);
+
+        // Assert
+        var diagnostic = Assert.Single(diagnostics);
+        Assert.Same(DiagnosticDescriptors.DoNotUseConfigureWebHostWithConfigureHostBuilder, diagnostic.Descriptor);
+        AnalyzerAssert.DiagnosticLocation(source.DefaultMarkerLocation, diagnostic.Location);
+        Assert.Equal("ConfigureWebHost cannot be used with WebApplicationBuilder.Host", diagnostic.GetMessage(CultureInfo.InvariantCulture));
+    }
+
+    [Fact]
+    public async Task WebApplicationBuilder_HostWithConfigureWebHostWithOptions_ProducesDiagnostics()
+    {
+        // Arrange
+        var source = TestSource.Read(@"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+var builder = WebApplication.CreateBuilder();
+builder.Host./*MM*/ConfigureWebHost(webHostBuilder => { }, optionsBuilder => { });
+");
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source.Source);
+
+        // Assert
+        var diagnostic = Assert.Single(diagnostics);
+        Assert.Same(DiagnosticDescriptors.DoNotUseConfigureWebHostWithConfigureHostBuilder, diagnostic.Descriptor);
+        AnalyzerAssert.DiagnosticLocation(source.DefaultMarkerLocation, diagnostic.Location);
+        Assert.Equal("ConfigureWebHost cannot be used with WebApplicationBuilder.Host", diagnostic.GetMessage(CultureInfo.InvariantCulture));
+    }
+
+    [Fact]
+    public async Task WebApplicationBuilder_WebHostWithConfigureWebHost_ProducesDiagnostics_In_Program_Main()
+    {
+        // Arrange
+        var source = TestSource.Read(@"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+public static class Program
+{
+    public static void Main(string[] args)
+    {
+        var builder = WebApplication.CreateBuilder(args);
+        /*MM*/builder.Host.ConfigureWebHost(webHostBuilder => { });
```

Comment:
Is the marker here supposed to be on the line start instead of at the problematic invocation?

Reply:
The pattern I've been following is to emit the diagnostic at the point where the problematic invocation exists so if the user clicks on the diagnostic's location in the build output the cursor will be directed to the place where they need to make the fix.

It also makes it easier to write codefixes associated with the analyzer because you can get the syntax at the problematic location right away.

Reply:
Yeah I figured that would be the case. I originally put it before `ConfigureWebHost`, but then the tests failed even though I was just using the syntax given to the analyzer to get the location for the invocation. I'll dig around more and see if I can work out why there's a mismatch when I do that so the warning is in the best place.